bip-0341.mediawiki

```diff
@@ -66,8 +66,10 @@ The following rules only apply when such an output is being spent. Any other out
 * If there are at least two witness elements left, script path spending is used:
 ** Call the second-to-last stack element ''s'', the script.
 ** The last stack element is called the control block ''c'', and must have length ''33 + 32m'', for a value of ''m'' that is an integer between 0 and 128<ref>'''Why is the Merkle path length limited to 128?''' The optimally space-efficient Merkle tree can be constructed based on the probabilities of the scripts in the leaves, using the Huffman algorithm. This algorithm will construct branches with lengths approximately equal to ''log<sub>2</sub>(1/probability)'', but to have branches longer than 128 you would need to have scripts with an execution chance below 1 in ''2<sup>128</sup>''. As that is our security bound, scripts that truly have such a low chance can probably be removed entirely.</ref>, inclusive. Fail if it does not have such a length.
+** Let ''c[0]'' be the ''control byte'' consisting of:
+*** Let ''v = c[0] & 0xfe'' and call it the ''leaf version''<ref>'''What constraints are there on the leaf version?''' First, the leaf version cannot be odd as ''c[0] & 0xfe'' will always be even, and cannot be ''0x50'' as that would result in ambiguity with the annex. In addition, in order to support some forms of static analysis that rely on being able to identify script spends without access to the output being spent, it is recommended to avoid using any leaf versions that would conflict with a valid first byte of either a valid P2WPKH pubkey or a valid P2WSH script (that is, both ''v'' and ''v | 1'' should be an undefined, invalid or disabled opcode or an opcode that is not valid as the first opcode). The values that comply to this rule are the 32 even values between ''0xc0'' and ''0xfe'' and also ''0x66'', ''0x7e'', ''0x80'', ''0x84'', ''0x96'', ''0x98'', ''0xba'', ''0xbc'', ''0xbe''. Note also that this constraint implies that leaf versions should be shared amongst different witness versions, as knowing the witness version requires access to the output being spent.</ref>.
+*** Let ''pyparity = c[0] & 0x01'', the parity of the Y coordinate of point ''Q''.
```

Comment:
_Q_ is used here before it's defined, and it doesn't make sense to include it in the definition of _pyparity_ since we only check later whether it's the parity of the Y coordinate of _Q_.

Also, what is "pyparity" supposed to mean? Why not _tweak_parity_ or just _parity_?

Reply:
"pyparity" was suggested [above](https://github.com/bitcoin/bips/pull/1329#discussion_r931037158) and I think was to suggest "public-key's y-parity".

A naked "parity" seems too generic.  "tweak_parity" seems a bit incomplete as a shorthand for "tweaked_public_key's_parity".  ??